src/ripple/app/main/Main.cpp

```diff
@@ -54,30 +54,47 @@ namespace po = boost::program_options;
 
 namespace ripple {
 
-void setupServer (Application& app)
+boost::filesystem::path
+getEntropyFile(Config const& config)
+{
+    auto const path = config.legacy("database_path");
+    if (path.empty ())
+        return {};
+    return boost::filesystem::path (path) / "random.seed";
+}
+
+bool
+adjustDescriptorLimit(int needed)
 {
 #ifdef RLIMIT_NOFILE
+    // Get the current limit, then adjust it to what we need.
     struct rlimit rl;
+
+    int available = 0;
+
     if (getrlimit(RLIMIT_NOFILE, &rl) == 0)
     {
-         if (rl.rlim_cur != rl.rlim_max)
-         {
-             rl.rlim_cur = rl.rlim_max;
-             setrlimit(RLIMIT_NOFILE, &rl);
-         }
+        available = rl.rlim_cur;
+
+        if (rl.rlim_cur != rl.rlim_max)
```

Comment:
Can we change this to something like:
if ((rl.rlim_cur < rl.rlim_max) || (rl.rlim_cur < needed))
That way, if the process has the right to exceed its hard limit (CAP_SYS_RESOURCE on Linux), it will work.